src/psyclone/psyir/backend/c.py

```diff
@@ -408,4 +413,40 @@ def codeblock_node(self, node):
         :raises VisitorError: The CodeBlock can not be translated to C.
 
         '''
+        # pylint: disable=no-self-use,unused-argument
         raise VisitorError("CodeBlocks can not be translated to C.")
+
+    @property
+    def do_loop_format(self):
+        '''Returns the format for a do/for loop. The format variables will
+        be replaced as follows:
+        0: indentation string
+        1: loop variable
+        2: first loop iteration
+        3: last loop iteration
+        4: step size
+        5: body of the loop
+
+        :return: the format of a loop.
+        :rtype: str
+        '''
+        return "{0}for({1}={2}-1; {1}<{3}; {1}+={4})\n"\
```

Comment:
I don't think there should be a `-1` in here, or, at least, we should document that this founction returns the first loop iteration minus one. I don't know enough about the C backend but maybe @sergisiso can comment - what's our strategy in generating C when thinking about the fact that array accesses are zero-indexed (and, consequently, loop bounds are adjusted c.f. those in Fortran)?

Reply:
See discussion below.